storage/storage.go

```diff
@@ -2076,6 +2076,23 @@ func (wb *withBackoff) apply(config *retryConfig) {
 	config.backoff = &wb.backoff
 }
 
+// WithMaxAttempts configures a maximum number of retries for potentially failing operations.
+// Without this setting, operations will continue retrying indefinitely
+// until either the context is canceled or a deadline is reached.
+func WithMaxAttempts(maxAttempts int) RetryOption {
+	return &withMaxAttempts{
+		maxAttempts: maxAttempts,
+	}
+}
+
+type withMaxAttempts struct {
+	maxAttempts int
+}
+
+func (wb *withMaxAttempts) apply(config *retryConfig) {
+	config.maxAttempts = wb.maxAttempts
+}
```

Comment:
As I said before, I think explicitly passing `WithMaxAttempts(0)` should mean no retries. We could implement this by setting the policy to `RetryNever` in the config if the user passed in 0 (since the default for an int field is 0, so we can't go based on that -- alternatively we could change the datatype to a `*int` internally, but I think using policy is simplest). Let's make this change and also add a test case for it in storage_test, and document the behavior of 0 in the doc for WithMaxAttempts

Reply:
What if someone did first `WithMaxAttempts(0)` and then `WithPolicy(RetryAlways)`. It will override the policy from `RetryNever` to `RetryAlways`